### PR TITLE
[EBP-112] Delete package servlet.

### PR DESCRIPTION
**Pull request by aliakseiTraihel**
https://github.com/exadel-inc/etoolbox-backpack/issues/112

### CODE REVIEW

File: core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/DeletePackageServlet.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.servlet.Servlet;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Serves as the network endpoint for user requests that delete package<br><br>
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/DeletePackageServlet.java:15]: (cosmetic) Serves as the network endpoint for user requests for deleting a package (or: to delete a package)
Also, do you need both `<br><br>` and `<p>` afterwards? Looks like to many spaces of different kinds all at once)

File: core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/DeletePackageServlet.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private static final long serialVersionUID = 1L;
+    private static final String PACKAGE_PATH_REQUEST_PARAMETER = "path";
+
+    @Reference
+    @SuppressWarnings("UnusedDeclaration")
+    private transient DeletePackageService deletePackageService;
+
+    /**
+     * Processes {@code DELETE} requests to the current endpoint. Attempts to delete a package and clear cache info about package
+     * Request parameters are parsed to a {@link String} with path to package and passed
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/DeletePackageServlet.java:40]: (cosmetic) with THE path to A package

File: core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/DeletePackageServlet.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+     * @param response {@code SlingHttpServletResponse} instance
+     */
+    @Override
+    protected void doDelete(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        RequestParameter requestParameter = request.getRequestParameter(PACKAGE_PATH_REQUEST_PARAMETER);
+        if (requestParameter != null) {
+            deletePackageService.delete(request.getResourceResolver(), requestParameter.getString());
+            response.setStatus(HttpServletResponse.SC_OK);
+        } else {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/DeletePackageServlet.java:53]: ❗ I  think HTTP status semantics is a bit violated here. 
If the request parameter is missing, the request is invalid. The user who issued such a request made an error. So this is 400. 
However is the request paramter is present, we must check if the specified package exists. If the package does not exist and could not be deleted, this is 404.
 If the package exists but one could not delete it (probably, due to the lack of rights), this is 403.
If the package exists, and we managed to delete it, this is 200.

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/DeletePackageService.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.exadel.etoolbox.backpack.core.services.pckg;
+
+import org.apache.sling.api.resource.ResourceResolver;
+
+/**
+ * Represents a service running in an AEM instance responsible for deleting package
+ */
+public interface DeletePackageService {
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/DeletePackageService.java:8]: (Possible improvement) Guys, is there a reason for having not a usual create-update-delete service but as many as five different services, one to create a package, one to edit, one to delete, etc.?

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/DeletePackageServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeletePackageServiceImpl.class);
+
+    @Reference
+    private BasePackageService basePackageService;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void delete(ResourceResolver resourceResolver, String path) {
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/DeletePackageServiceImpl.java:32]: I think this method must have a return value depending on the actual result of the operation (see also my very last comment down below)

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/DeletePackageService.java (hunk 4 above)
Comment (aliakseiTraihel) [reply core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/DeletePackageService.java:8]: may be not, we can use one

**Review by liubou-masiuk — APPROVED** [review]

**Pull request merged**